api/client.js

```diff
@@ -61,12 +61,14 @@ const Admin = require('./facades/admin-v3.js');
       see <https://www.npmjs.com/package/macaroon-bakery>;
     - closeCallback: a callback to be called with the exit code when the
       connection is closed.
-  @param {Function} callback Called when the connection is made, the callback
-    receives an error and a client object. If there are no errors, the client
-    can be used to login and logout to Juju. See the docstring for the Client
-    class for information on how to use the client.
+  @param {Function} [callback=null] Called when the connection is made, the
+    callback receives an error and a client object. If there are no errors, the
+    client can be used to login and logout to Juju. See the docstring for the
+    Client class for information on how to use the client.
+  @return {Promise} This promise will be rejected if there is an error connecting,
```

Comment:
Please mention that the promise will not be resolved or rejected if a callback is provided.